packages/react-sdk/src/components/slot.tsx

```diff
@@ -0,0 +1,17 @@
+import { forwardRef, type ElementRef, type ReactNode } from "react";
+
+type Props = {
+  children?: ReactNode;
+};
+
+export const Slot = forwardRef<ElementRef<"div">, Props>((props, ref) => {
+  return (
+    <div
+      {...props}
+      ref={ref}
+      style={{ display: props.children ? "contents" : "block" }}
```

Comment:
maybe we could set contents on fragment, so we can always have slot selectable? its def. inconsistent to not be able to select the slot, especially once we want to add props to it

Reply:
oh I forgot why we needed display: contents - it was because we didn't want the slot to interfere with the layout

I guess we need some other way to  get selection working then

Reply:
ok here is what we can do:
1. try getting getClientRects()
2. if length is 0 but element has children, 
3. walk children calculate their group height/width
4. in case of fragment its also going to have length 0, so we would need to do this in a loop to avoid having specific logic for slots and fragments

I think we should do it in a way that allows us outlining of elements even when they have display: contents in a generic way, without using the logic of fragment/slot component.

This way we can outline even when user component uses this and we don't know anything about it